inetdiag.Parse and message cache

gfr10598 opened the pull request:
Adds basic inet message and attribute parsing, and a message cache.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-lab/tcp-info/11)
<!-- Reviewable:end -->

pboothe commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, small comments.

---

Review status: 0 of 7 files reviewed at latest revision, all discussions resolved.

---

*[cache/cache.go, line 22 at r2](https://reviewable.io/reviews/m-lab/tcp-info/11#-LCVDcsc98FlnWz_H0cv:-LCVDcsc98FlnWz_H0cw:b-7de7wf) ([raw file](https://github.com/m-lab/tcp-info/blob/9af48b07148a3d55fe05b49a0df1f4264cece240/cache/cache.go#L22)):*
> ```Go
> 	// Map from inode to ParsedMessage
> 	current map[uint32]*inetdiag.ParsedMessage
> 	past    map[uint32]*inetdiag.ParsedMessage
> ```

"stale" might be better than "past". It doesn't contain the inodes from the last round of updates, it contains the inodes from the last round that did not get updated this round.

---

*[cache/cache.go, line 31 at r2](https://reviewable.io/reviews/m-lab/tcp-info/11#-LCVCKBi9a6Z8zhrO7oR:-LCVCKBi9a6Z8zhrO7oS:buskouk) ([raw file](https://github.com/m-lab/tcp-info/blob/9af48b07148a3d55fe05b49a0df1f4264cece240/cache/cache.go#L31)):*
> ```Go
> }
> 
> // Update swaps msg with the cache contents, and returns the cached value.
> ```

...and returns the evicted value from the cache.

---

*[inetdiag/inetdiag_test.go, line 155 at r2](https://reviewable.io/reviews/m-lab/tcp-info/11#-LCVCr6v0AuSnyVJFfqx:-LCVCr6v0AuSnyVJFfqy:b-lk4pif) ([raw file](https://github.com/m-lab/tcp-info/blob/9af48b07148a3d55fe05b49a0df1f4264cece240/inetdiag/inetdiag_test.go#L155)):*
> ```Go
> 
> func TestParseGarbage(t *testing.T) {
> 	var json1 = `{"Header":{"Len":356,"Type":20,"Flags":2,"Seq":1,"Pid":148940},"Data":"CgEAAOpWE6cmIAAAEAMEFbM+nWqBv4ehJgf4sEANDAoAAAAAAAAAgQAAAAAdWwAAAAAAAAAAAAAAAAAAAAAAAAAAAAC13zIBBQAIAAAAAAAFAAUAIAAAAAUABgAgAAAAFAABAAAAAAAAAAAAAAAAAAAAAAAoAAcAAAAAAICiBQAAAAAAALQAAAAAAAAAAAAAAAAAAAAAAAAAAAAArAACAAEAAAAAB3gBQIoDAECcAABEBQAAuAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAUCEAAAAAAAAgIQAAQCEAANwFAACsywIAJW8AAIRKAAD///9/CgAAAJQFAAADAAAALMkAAIBwAAAAAAAALnUOAAAAAAD///////////ayBAAAAAAASfQPAAAAAADMEQAANRMAAAAAAABiNQAAxAsAAGMIAABX5AUAAAAAAAoABABjdWJpYwAAAA=="}`
> ```

I can't tell the difference between the good input data and the bad input data. Is that okay? Maybe. But I thought I should let you know.

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/tcp-info/11#-:-LCVDw7B7bMMMUH3L170:b-2xesp0)*
<!-- Sent from Reviewable.io -->

gfr10598 commented:
Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[cache/cache.go, line 22 at r2](https://reviewable.io/reviews/m-lab/tcp-info/11#-LCVDcsc98FlnWz_H0cv:-LCVXctS6mSBROVrQCOq:b-ij67z8) ([raw file](https://github.com/m-lab/tcp-info/blob/9af48b07148a3d55fe05b49a0df1f4264cece240/cache/cache.go#L22)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

"stale" might be better than "past". It doesn't contain the inodes from the last round of updates, it contains the inodes from the last round that did not get updated this round.
</blockquote></details>

It initially contains all existing connections, as of last cycle.  Connections are removed as they are replaced in the new cycle.
Changed name and added comments that hopefully make this clearer.

---

*[cache/cache.go, line 31 at r2](https://reviewable.io/reviews/m-lab/tcp-info/11#-LCVCKBi9a6Z8zhrO7oR:-LCVXtdn5QMG4GoPXbVT:b-896fix) ([raw file](https://github.com/m-lab/tcp-info/blob/9af48b07148a3d55fe05b49a0df1f4264cece240/cache/cache.go#L31)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

...and returns the evicted value from the cache.
</blockquote></details>

Done.

---

*[inetdiag/inetdiag_test.go, line 155 at r2](https://reviewable.io/reviews/m-lab/tcp-info/11#-LCVCr6v0AuSnyVJFfqx:-LCVYY1j1amO61oDpQNs:boxhkkr) ([raw file](https://github.com/m-lab/tcp-info/blob/9af48b07148a3d55fe05b49a0df1f4264cece240/inetdiag/inetdiag_test.go#L155)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary><blockquote>

I can't tell the difference between the good input data and the bad input data. Is that okay? Maybe. But I thought I should let you know.
</blockquote></details>

Improved comments.

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/tcp-info/11)*
<!-- Sent from Reviewable.io -->
